crates/completion/src/completions/pattern.rs

```diff
@@ -15,20 +17,25 @@ pub(crate) fn complete_pattern(acc: &mut Completions, ctx: &CompletionContext) {
     // suggest variants + auto-imports
     ctx.scope.process_all_names(&mut |name, res| {
         let add_resolution = match &res {
-            hir::ScopeDef::ModuleDef(def) => {
-                if ctx.is_irrefutable_let_pat_binding {
-                    matches!(def, hir::ModuleDef::Adt(hir::Adt::Struct(_)))
-                } else {
-                    matches!(
-                        def,
-                        hir::ModuleDef::Adt(hir::Adt::Enum(..))
-                            | hir::ModuleDef::Adt(hir::Adt::Struct(..))
-                            | hir::ModuleDef::Variant(..)
-                            | hir::ModuleDef::Const(..)
-                            | hir::ModuleDef::Module(..)
-                    )
+            hir::ScopeDef::ModuleDef(def) => match def {
+                hir::ModuleDef::Adt(hir::Adt::Struct(strukt)) => {
+                    acc.add_struct_pat(ctx, strukt.clone(), Some(name.clone()));
+                    true
+                }
+                hir::ModuleDef::Variant(variant)
+                    if !ctx.is_irrefutable_pat_binding
+                        // render_resolution already does some pattern completion tricks for tuple variants
```

Comment:
`render_resolution` already does pattern completions like this for variants like Option::Some, this feature collides with that in a test because the test finds multiple completions there which is why for now I just disabled this full completion for tuple variants. It would probably make sense to replace that one in the end?

Reply:
Hm, so we have two cases where we want to complete args of `Some(_)` -- when we use it as a value `let x = Some(92)` and when we use it as a pattern `if let Some(92) = x`. I think that pattern position should be implemented by the code in this PR, while function position should probably be kept where it is now.